collector/utils.go

```diff
@@ -0,0 +1,176 @@
+// +build linux
+
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2017 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collector
+
+import (
+	"fmt"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/intelsdi-x/snap-plugin-lib-go/v1/plugin"
+
+	"github.com/intelsdi-x/snap-plugin-collector-docker/container"
+)
+
+// dynamicElement is defined by its name and description
+type dynamicElement struct {
+	name        string
+	description string
+}
+
+type nsCreator struct {
+	dynamicElements map[string]dynamicElement
+}
+
+// definedDynamicElements holds expected dynamic element(s) with definition in docker metrics namespaces which occurs after the key-word
+var definedDynamicElements = map[string]dynamicElement{
+	"filesystem":                 {"device_name", "a name of filesystem device"},
+	"labels":                     {"label_key", "a key of container's label"},
+	"network":                    {"network_interface", "a name of network interface or 'total' for aggregate"},
+	"per_cpu":                    {"cpu_id", "an id of cpu"},
+	"io_service_bytes_recursive": {"device_name", "a name of block device"},
+	"io_serviced_recursive":      {"device_name", "a name of block device"},
+	"io_queue_recursive":         {"device_name", "a name of block device"},
+	"io_service_time_recursive":  {"device_name", "a name of block device"},
+	"io_wait_time_recursive":     {"device_name", "a name of block device"},
+	"io_merged_recursive":        {"device_name", "a name of block device"},
+	"io_time_recursive":          {"device_name", "a name of block device"},
+	"sectors_recursive":          {"device_name", "a name of block device"},
+	"hugetlb_stats":              {"size", "hugetlb page size"},
+}
+
+func initClient(c *collector, endpoint string) error {
+	dc, err := container.NewDockerClient(endpoint)
+	if err != nil {
+		return err
+	}
+
+	c.cgroupfs, err = dc.GetDockerParam("CgroupDriver")
+	if err != nil {
+		return err
+	}
+
+	drv, err := dc.GetDockerParam("Driver")
```

Comment:
Consider change it to `c.driver, err = dc.GetDockerParam("Driver")`, so it will exactly the same way how it's done above`